### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by ShoiraTa**
# Game Logic: Milestone 3
Added game logic as per milestone 3 requirement,  built the game logic by adding different classes in a different file, and included them in the main file.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  def win?(board)
+    @win_combination.any? do |combination|
+      combination.all? { |idx| board[idx] == 'X' } || combination.all? { |idx| board[idx] == 'O' }
+    end
+  end
+
+  def position_valid?(board, index)
+    if board[index] == 'X' || board[index] == 'O'
+      puts
+      puts 'This position is taken, choose another position'
```
Comment (umairarshadbutt) [inline lib/board.rb]: Move these puts into `main.rb`

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  def name_valid?(name)
+    true if name.length >= 3 && !name.nil?
+  end
+
+  def player_input_valid?(player_input)
+    if player_input.between?(0, 9) && !player_input.nil?
+      true
+    else
+
+      p 'input a valid number'
```
Comment (umairarshadbutt) [inline lib/player.rb]: Move this `p` into `main.rb`

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @ShoiraTa 
Such a remarkable project you made :+1: !
We, however, do have a few changes to carry out before we proceed.

##

Kindly work on the suggested changes  👇.
- [x] there should be no `puts/print` and `gets.chomp` outside `main.rb` file.([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
- [ ] All game-logic-related resources (classes and/or modules) are kept in the lib directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by ShoiraTa** [comment]
Hello, 
Thank you for your comments, it was quite clear for further actions. I have moved all puts/print into main.rb file.

**Comment by mimipeshy** [comment]
## STATUS: APPROVED :dancing_women: :partying_face:
Hello team @ShoiraTa   👋🏾 👋🏾 congratulations on doing such a  job on this project

#### Highlights :white_check_mark: :white_check_mark:
- [ ]  Your project works flawlessly and have gone the extra mile to add some validations 👏🏾 
- [ ]  You have worked on the TSE comments given ✔️ 
- [ ] You have incorporated linters

**[OPTIONAL]**

- [ ] When a user makes a wrong move for example enters a string this error `"-1 is not valid, please input a valid position"` you can either make your project to be smart enough to read the input or just leave a plain message `sorry, that is an invalid move kindly choose moves from 1-9`. 😄  this way the error messages are clear for a user.

With that, I approve of this milestone. All the best in the next projects :technologist:
Regards
Peris Ndanu

**Pull request merged**